packages/flutter/test/widgets/image_test.dart

```diff
@@ -1323,3 +1485,24 @@ class TestImage implements ui.Image {
   @override
   String toString() => '[$width\u00D7$height]';
 }
+
+class DebouncingImageProvider extends ImageProvider<Object> {
+  DebouncingImageProvider(this.imageProvider);
+
+  static Set<Object> seenKeys = <Object>{};
```

Comment:
Why is this static? How does it get reset between tests? Could this just be on the instance to avoid leakage of state between tests?

Reply:
The idea is that this is a cache that has to be shared between at least two instances.  But it should get reset between tests.

I could easily reset this between tests - do you think that's good enough?

Reply:
That works. Alternatively (and if this wouldn't be a test), DebouncingImageProvider could just take a cache object to make it explicit that it's sharing state between instances..

Reply:
Oh, and maybe add a comment to this property explaining why it is static for the next reader?

Reply:
Made this be passed into the provider to make it clearer, and added comment.